src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,470 @@
+    sleep 10
+    date
+    exit
+done
+~~~
+
+Next, make the script file executable:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+chmod 777 wal-flip.sh
+~~~
+
+Next, cause a long stall:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+long-stall.sh
+~~~
+
+#### WAL Failover metrics
+
+{% include {{ page.version.version }}/wal-failover-metrics.md %}
+
+Whenever a WAL failover occurs on a disk, `wal.failover.switch.count` for the associated store will increment by 1.
+
+When the disk continues to be stalled for longer than the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables), the node goes down, and there is no more metrics data coming from that node.
+
+You can confirm that the node is down (no longer live and available) using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+## Summary
+
+### Important environment variables
+
+Variable | Description | Default | Recommended value with WAL failover enabled
+---------|-------------|---------|---------------------------------------------
+`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT` | The threshold above which an observed engine sync duration triggers a fatal error. This environment variable is the default for the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting. If that cluster setting is explicitly set, it takes precedence over this environment variable. | `20s`   | `40s`
+`COCKROACH_LOG_MAX_SYNC_DURATION` | The maximum duration the file sink is allowed to take to write a log entry before we fatal the process for a disk stall. | `20s` | `40s`
+
+### To enable WAL Failover
+
+[XXX](): YOU ARE HERE
+
+Setup your cluster for WAL failover with either [multiple stores](#multi-store-config) or a [side disk](#single-store-config).
+
+For [multiple stores](#multi-store-config), pass `--wal-failover=among-stores` to [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}).
+
+For a [side disk on a single store config](#single-store-config), pass `--wal-failover={ path-to-side-drive-directory }` to [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}).
+
+Use remote log sinks, or if you use file-based logging, enable asynchronous buffering of `file-groups` log sinks:
+
+{% include {{ page.version.version }}/misc/logging-flags.md %}
+
+Change the value of `COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT` by setting it as follows:
+
+1. Before starting the cluster, set it using the [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables) environment variable.
+2. After starting the cluster, set it by changing the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting.
+
```

Comment:
Maybe remove this line break and make the below lines part of #2 here.

Reply:
✅